### PR TITLE
Save preferred theme in localStorage

### LINKED ISSUE #6242: add possibility to switch dark/light theme
# Description

## User Stories

* > As a user, I want the web UI to respect my preferences regarding light/dark mode

* > As a user, I want to be able to switch between light/dark mode and would like the browser to save my settings

* > As an admin, I would like to provide my users with a customized/branded experience both in light and dark mode

## Acceptance Criteria
- [x] Add dark default theme to theme.json
- [x] Save all themes from theme.json to Vuex store
- [x] Initialize web UI with user's browser preference regarding light/dark mode
- [x] Add theme switcher button to top bar
- [ ] Save user's settings in localStorage
- [ ] Update docs so admin users can customize both their light & dark theme

## Definition of done
- Functional requirements
[ ] functionality described in the user story works
[ ] acceptance criteria are fulfilled
- Quality
[ ] code review happened
[ ] CI is green
[ ] critical code received unit tests by the developer
[ ] automated tests passed (if automated tests are not available, this test needs to be created and passed
- Non-functional requirements
[ ] no sonar cloud issues

### PR DESCRIPTION
**Pull request by pascalwengerter**
## Description
~~Code doesn't read that nicely but does what it should~~ (comment from before we switched to composition api)

## Related Issue
- Fixes #6242

### CODE REVIEW

File: packages/web-runtime/src/components/Topbar/ThemeSwitcher.vue [hunk 1]
```diff
@@ -36,20 +36,33 @@ export default {
     }
   },
   mounted() {
-    if (window.matchMedia('(prefers-color-scheme: dark)').matches) {
+    const currentTheme = JSON.parse(window.localStorage.getItem('oc_currentTheme'))
```
Comment (kulmann) [inline packages/web-runtime/src/components/Topbar/ThemeSwitcher.vue:39]: We have a `useLocalStorage` composable, what about using that one?

Comment (pascalwengerter) [reply packages/web-runtime/src/components/Topbar/ThemeSwitcher.vue:39]: Interesting, let me take a look 🕵🏽 only searched the codebase for `localStorage` and got "inspired" ™️  by what I found

Comment (pascalwengerter) [reply packages/web-runtime/src/components/Topbar/ThemeSwitcher.vue]: Refactored, thanks for the input 🙌🏽

**Review by kulmann — APPROVED** [review]
A bit fishy that I approve this, but LGTM 👍

**Comment by ownclouders** [comment]
Results for <strong>oC10iPhone3</strong> https://drone.owncloud.com/owncloud/web/22371/49/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Pull request merged**